### PR TITLE
Add international phone input for business contact form

### PR DESCRIPTION
**Pull request by ania-hm**
## Pull Request Summary

We want to use the intl-tel-input JavaScript plugin for better user experience during phone number input.

## Feedback

N/A

### Before usage the intl-tel-input

phone-input-without-intl-tel-input

### After usage the intl-tel-input

phone-number-with-intl-tel-input

### CODE REVIEW

File: app/views/business_contact_forms/new.slim [hunk 1]
```diff
@@ -1,3 +1,7 @@
 - content_for :body_class, 'business-contact-forms'
+- content_for :custom_heads do
+  = stylesheet_link_tag("https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/18.3.3/css/intlTelInput.css", integrity: "sha512-0MvDK6EKxNTNdRSbB3hF2+1kLkJgC9yHVDf//8hPfEI2cDfy/Mezjw9BsSyA61uX2Vkk9iDOJAHoBToust2lsw==", crossorigin: "anonymous", referrerpolicy: "no-referrer")
+  = stylesheet_link_tag("intl-tel-input")
```
Comment (torrocus) [inline app/views/business_contact_forms/new.slim:4]: Please check if we can maintain the existing standard.
```suggestion
  = stylesheet_link_tag('https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/18.3.3/css/intlTelInput.css', integrity: 'sha512-0MvDK6EKxNTNdRSbB3hF2+1kLkJgC9yHVDf//8hPfEI2cDfy/Mezjw9BsSyA61uX2Vkk9iDOJAHoBToust2lsw==', crossorigin: 'anonymous', referrerpolicy: 'no-referrer')
  = stylesheet_link_tag('intl-tel-input')
```

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -32,6 +32,7 @@
   <!-- <base href='https://fractalsoft.org/' /> -->
   <%= favicon_link_tag '/favicon.ico', skip_pipeline: true %>
   <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload', async: Rails.env.production? %>
+  <%= yield(:custom_heads) %>
```
Comment (torrocus) [inline app/views/layouts/application.html.erb:35]: I'm thinking about this name and what comes to mind is:
```suggestion
  <%= yield(:custom_stylesheets) %>
```

File: app/javascript/application.js [hunk 3]
```diff
@@ -8,3 +8,4 @@ import "@popperjs/core";
 import "bootstrap";
 import "fractalsoft/browser_info";
 import "fractalsoft/contact_forms";
+import "fractalsoft/phone";
```
Comment (torrocus) [inline app/javascript/application.js:11]:
```suggestion
import "fractalsoft/phone";

```

File: app/assets/stylesheets/intl-tel-input.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  height: 56px;
+  &__country-list {
+    // popup so render above everything else
+    z-index: 1060;
+  }
+}
+
+.contact-phone-input {
+  height: 56px;
+}
```
Comment (torrocus) [inline app/assets/stylesheets/intl-tel-input.scss:25]: Shouldn't this be inside `.iti` class?

File: app/javascript/fractalsoft/phone.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  var iti = intlTelInput(input_phone, {
+    separateDialCode: true,
+    initialCountry: "auto",
+    geoIpLookup: function(callback) {
+      fetch("https://ipapi.co/json")
+        .then(function(res) { return res.json(); })
+        .then(function(data) { callback(data.country_code); })
+        .catch(function() { callback("us"); });
+    }
```
Comment (torrocus) [inline app/javascript/fractalsoft/phone.js:16]: nice!

File: app/views/business_contact_forms/_form.slim [hunk 6]
```diff
@@ -14,7 +14,9 @@
           .col-lg-6.col-12
             = f.input :email, placeholder: t('mail_form.attributes.business_contact_form.email'), required: true
           .col-lg-6.col-12
-            = f.input :phone_number, placeholder: t('mail_form.attributes.business_contact_form.phone_number')
+            = f.input :phone_number, type: 'tel', label: false, placeholder: t('mail_form.attributes.business_contact_form.phone_number'), input_html: { class: 'js-phone-input contact-phone-input' }
+            = f.input :phone_country_code, as: :hidden, input_html: { class: 'js-country-code' }
+            = f.input :phone_country_iso2, as: :hidden, input_html: { class: 'js-country-iso2' }
```
Comment (torrocus) [inline app/views/business_contact_forms/_form.slim:19]: A small proposal to change the name to make it clear what it is for.
```suggestion
            = f.input :phone_country_code, as: :hidden, input_html: { class: 'js-phone-country-code' }
            = f.input :phone_country_iso2, as: :hidden, input_html: { class: 'js-phone-country-iso2' }
```

File: app/views/business_contact_forms/_form.slim [hunk 7]
```diff
@@ -3,7 +3,7 @@
     = t('.title')
   .row.justify-content-center
     .col-md-8.col-11
-      = simple_form_for @business_contact_form do |f|
+      = simple_form_for @business_contact_form, html: { id: 'contact_form' } do |f|
```
Comment (torrocus) [inline app/views/business_contact_forms/_form.slim:6]: Doesn't the form have its own id? Just asking.

File: app/javascript/fractalsoft/phone.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  input_country_code.value = iti.getSelectedCountryData().dialCode;
+  input_country_iso2.value = iti.getSelectedCountryData().iso2;
+
+  input_phone.addEventListener('countrychange', function(e) {
+  input_country_code.value = iti.getSelectedCountryData().dialCode;
+  input_country_iso2.value = iti.getSelectedCountryData().iso2;
+  });
```
Comment (torrocus) [inline app/javascript/fractalsoft/phone.js:25]: I think you've lost the indentation here.
```suggestion
  input_phone.addEventListener('countrychange', function(e) {
    input_country_code.value = iti.getSelectedCountryData().dialCode;
    input_country_iso2.value = iti.getSelectedCountryData().iso2;
  });
```

File: app/javascript/fractalsoft/phone.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    geoIpLookup: function(callback) {
+      fetch("https://ipapi.co/json")
+        .then(function(res) { return res.json(); })
+        .then(function(data) { callback(data.country_code); })
+        .catch(function() { callback("us"); });
+    }
+  });
+
+  input_country_code.value = iti.getSelectedCountryData().dialCode;
+  input_country_iso2.value = iti.getSelectedCountryData().iso2;
```
Comment (torrocus) [inline app/javascript/fractalsoft/phone.js:20]: When I look at it, I see duplication. Maybe we should trigger the `countrychange` event somehow to start with?

**Review by torrocus — COMMENTED** [review]
Overall, it looks good. I added my two cents.

File: app/views/business_contact_forms/_form.slim (hunk 7 above)
Comment (ania-hm) [reply app/views/business_contact_forms/_form.slim:6]: It has its own id. Maybe I think in wrong direction, but for other contact forms we have id `contact-form`. I added the same id if in the future we want to add phone number in other contact forms. Then we will use the same intl-tel-input.scss. Maybe this is wrong thinking, over-something.

**Review by torrocus — APPROVED** [review]

**Pull request merged**